TaskWebApp/Controllers/TasksController.cs

```diff
@@ -26,18 +27,26 @@ public async Task<ActionResult> Index()
             {
                 // Retrieve the token with the specified scopes
                 var scope = new string[] { Startup.ReadTasksScope };
-                string signedInUserID = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
-                TokenCache userTokenCache = new MSALSessionCache(signedInUserID, this.HttpContext).GetMsalCacheInstance();
-                ConfidentialClientApplication cca = new ConfidentialClientApplication(Startup.ClientId, Startup.Authority, Startup.RedirectUri, new ClientCredential(Startup.ClientSecret), userTokenCache, null);
 
-                var user = cca.Users.FirstOrDefault();
-                if (user == null)
+                
+                string signedInUserID = ClaimsPrincipal.Current.FindFirst(Startup.ObjectIdElement)?.Value;
+
+
+                ConfidentialClientApplication cca = Startup.GetConfidential();
+                TokenCacheHelper.EnablePersistence(cca.UserTokenCache);
+
+                var accounts = cca.GetAccountsAsync().Result;
+                var account = Startup.GetAccountByPolicy(accounts, Startup.DefaultPolicy)
+                                        ?? accounts.FirstOrDefault();
```

Comment:
Super dangerous !
A Web app is multi-users, and therefore as written (and because the cache used is not one per account), any user using the Web App would have access to the Web api with the creds of the first user in the cache!

Reply:
This is one of the reason why we strongly advise to use the ITokenProvider abstraction which is presented in the https://github.com/Azure-Samples/ms-identity-aspnetcore-webapp-tutorial: it handles everthing for you